### PR TITLE
NP-380 release script

### PR DESCRIPTION
**Pull request by panosvoudouris**
Fix the release script as we now can't push straight to the `master` branch. This creates two scripts, one for preparing the release (`release`) and one that is automatically run when trying to publish using `npm publish`.

To make release you run `npm run release`. This prepares the release and puts it in a branch that the needs a PR to be merged into master. Once that is merged to do the actual release switch to master and run `npm publish`. The `prePublish` script will ensure you can only run `npm publish` from a master that is in a clean state. There is still the risk of the new version branch being merged via the PR and another PR being merged before the release is done. This could be prevented if the script were to check for the changelog or git log history but that is probably overkill.

### CODE REVIEW

File: scripts/release.js [hunk 1]
```diff
@@ -197,78 +136,84 @@ prompt([
...
+                            git.add('.')
+                                .commit(releaseMessage)
+                                .push('origin', newVersionBranch, gitErr2 => {
+                                    if (gitErr2) {
+                                        showError(gitErr2, true);
+                                    }
+
+                                    log(
+                                        chalk.green(
+                                            `${ok} Changes commited and pushed. You must now create a pull request for branch v${newVersion}.`
```
Comment (shahzainb) [inline scripts/release.js:200]: `committed`?

File: scripts/release.js [hunk 2]
```diff
@@ -197,78 +136,84 @@ prompt([
...
+                                        )
+                                    );
+                                });
+                        } else {
+                            log(
+                                chalk.bold(
+                                    'Changes not commited. Check the repo status.'
```
Comment (shahzainb) [inline scripts/release.js:207]: `committed`?

File: scripts/releaseHelpers.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            err = true;
+        }
+
+        if (ahead !== 0) {
+            showError(`${error} Your repo has changes not pushed to remote.`);
+            err = true;
+        }
+
+        if (dirty !== 0) {
+            showError(`${error} Your have uncommited changes in you repo.`);
```
Comment (shahzainb) [inline scripts/releaseHelpers.js:40]: `You`, `uncommitted`, `your`

File: scripts/releaseHelpers.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                `${error} There ${
+                    untracked === 1 ? 'is' : 'are'
+                } ${untracked} untracked file${untracked === 1 ? '' : 's'}.`
+            );
+            err = true;
+        }
+
+        if (err) {
+            showError(
+                'Please ensure you are in the master branch and the the repo is in a clean state.',
```
Comment (shahzainb) [inline scripts/releaseHelpers.js:55]: `that the`

**Comment by shahzainb** [comment]
Can we also update the CONTRIBUTING.md to reflect the new release process as in the description of this PR? Thanks!

File: scripts/releaseHelpers.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        if (dirty !== 0) {
+            showError(`${error} Your have uncommited changes in you repo.`);
+            err = true;
+        }
+
+        if (untracked !== 0) {
+            showError(
+                `${error} There ${
+                    untracked === 1 ? 'is' : 'are'
```
Comment (baseonmars) [inline scripts/releaseHelpers.js:47]: 💅

File: scripts/releaseHelpers.js (hunk 3 above)
Comment (panosvoudouris) [reply scripts/releaseHelpers.js:40]: Spelling police

**Pull request merged**